hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MethodProbe.java

```diff
@@ -59,19 +59,20 @@
     }
 
     void register(MetricsRegistryImpl metricsRegistry, Object source, String namePrefix) {
-        String name = getName(namePrefix);
+        String name = namePrefix + '.' + getProbeOrMethodName();
         metricsRegistry.registerInternal(source, name, probe.level(), this);
     }
 
-    private String getName(String namePrefix) {
-        String name;
-        if (probe.name().equals("")) {
-            name = getterIntoProperty(method.getName());
-        } else {
-            name = probe.name();
-        }
+    void register(ProbeBuilderImpl builder, Object source) {
+        builder
+                .withTag("unit", probe.unit().name().toLowerCase())
```

Comment:
Mind that this is subject to change as unit is going to be a (value) field and not a tag.

Reply:
What is "value field"? It's fixed for a metric, can be part of the key, not value.

Reply:
Could this be solved by adding an explicit 'withUnit(UnitEnum e)' to the builder?